modules/tinymce/src/themes/silver/demo/ts/demo/ButtonSetupDemo.ts

```diff
@@ -78,6 +80,112 @@ export default {
+          type: 'menuitem',
+          text: info.title ?? info.value,
+          onAction: () => {
+            console.log('Triggering: ' + info.value);
+          }
+        });
+
+        const makeCategory = (title: string, items: any[]) => ({
+          type: 'nestedmenuitem',
+          text: title,
+          getSubmenuItems: () => items
+        });
+
+        const currentDateMerge = {
+          value: 'Current.Date',
+          title: 'Current date in DD/MM/YYYY format'
+        };
+
+        const tocMerge = {
+          value: 'Campaign.Toc',
+          title: 'Linked table of contents in your campaign'
+        };
+
+        const phoneHomeMerge = { value: 'Phone.Home' };
+        const phoneWorkMerge = { value: 'Phone.Work' };
+
+        const personFirstnameMerge = { value: 'Person.Name.First' };
+        const personSurnameMerge = { value: 'Person.Name.Last' };
+        const personFullnameMerge = { value: 'Person.Name.Full' };
+
+        const personWorkEmail = { value: 'Person.Email.Work' };
+        const personHomeEmail = { value: 'Person.Email.Home' };
+
+        if (!fetchContext || (fetchContext && fetchContext.pattern.length === 0)) {
+          callback([
+            makeMailMerge(currentDateMerge),
+            makeMailMerge(tocMerge),
+            makeCategory(
+              'Phone',
+              [
+                makeMailMerge(phoneHomeMerge),
+                makeMailMerge(phoneWorkMerge)
+              ]
+            ),
+            makeCategory(
+              'Person',
+              [
+                makeMailMerge(personFirstnameMerge),
+                makeMailMerge(personSurnameMerge),
+                makeMailMerge(personFullnameMerge),
+                makeCategory(
+                  'Email',
+                  [
+                    makeMailMerge(personWorkEmail),
+                    makeMailMerge(personHomeEmail)
+                  ]
+                )
+              ]
+            )
+          ] as any);
+        } else {
+          const allMerges: Array<{value: string; title?: string}> = [
+            currentDateMerge,
+            tocMerge,
+            phoneHomeMerge,
+            phoneWorkMerge,
+            personFirstnameMerge,
+            personSurnameMerge,
+            personFullnameMerge,
+            personWorkEmail,
+            personHomeEmail
+          ];
+
+          const matches = Arr.filter(allMerges, (m): boolean => {
+            const valueMatches = m.value.toLowerCase().indexOf(fetchContext.pattern.toLowerCase()) > -1;
+            return valueMatches || (
+              m.title !== undefined && (m.title.toLowerCase().indexOf(fetchContext.pattern.toLowerCase()) > -1)
+            );
+          });
+
+          if (matches.length > 0) {
+            callback(
+              Arr.map(matches, makeMailMerge) as any
+            );
+          } else {
+            callback([
+              {
+                type: 'menuitem',
+                text: 'No Results',
```

Comment:
Hmm. I wonder if we need something like this for the mergetags feature. We haven't any string for translation though. Should we just feed it an empty array if there are no results?

Reply:
Should this be handled by the search feature of the menubutton. It should not really be a button but rather a message that nothing matched and that should have some announce aria role maybe?

Reply:
Not sure. @vpyshnenko did we include a string for this in the mergetags plugin?

Reply:
We haven't included it. But probably we forgot to do it since this case was not represented in the design specification. In early versions of spec we had `No matches` message for empty automcompleter result, but later it was removed. It seems to me it's a good idea to make it customizable it for a specific context.